### ISSUE TITLE
File lock

### ISSUE THREAD

**Issue by pkanter** [innoverio/vscode-dbt-power-user#334]
### Expected behavior

I run `dbt clean` and `dbt deps` to update my packages.  Updates happen without error.

### Actual behavior

When I run `dbt clean` and then `dbt deps`, it appears that there's a conflict with the dbt_project_evaluator package.  I've not seen this previously.

### Steps To Reproduce

With `dbt power user` enabled, when I run `dbt clean` and then `dbt deps`, it appears that there's a conflict with the dbt_project_evaluator package.  I've not seen this previously.
If I disable `power user`, and then run `dbt deps`, my packages install as expected.  I then enable `dbt power user` and go on with my day.

### Log output/Screenshots

```
  File "c:\users\...\fivetran_dbt_integrated\.venv\lib\site-packages\dbt\deps\base.py", line 117, in _install
    connection_exception_retry(download_untar_fn, 5)
  File "c:\users\...\fivetran_dbt_integrated\.venv\lib\site-packages\dbt\utils.py", line 616, in _connection_exception_retry
    return fn()
  File "c:\users\...\fivetran_dbt_integrated\.venv\lib\site-packages\dbt\deps\base.py", line 129, in download_and_untar
    system.untar_package(tar_path, deps_path, package_name)
  File "c:\users\...\fivetran_dbt_integrated\.venv\lib\site-packages\dbt\clients\system.py", line 497, in untar_package
    dbt.clients.system.rename(downloaded_path, desired_path, force=True)
  File "c:\users\...\fivetran_dbt_integrated\.venv\lib\site-packages\dbt\clients\system.py", line 485, in rename
    shutil.move(from_path, to_path)
  File "C:\Users\...\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 578, in move
    rmtree(src)
  File "C:\Users\...\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 516, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "C:\Users\...\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 395, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Users\...\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 404, in _rmtree_unsafe
    onerror(os.rmdir, path, sys.exc_info())
  File "C:\Users\...\AppData\Local\Programs\Python\Python37\lib\shutil.py", line 402, in _rmtree_unsafe
    os.rmdir(path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'dbt_packages\\dbt-project-evaluator-0.4.1\\integration_tests'
```

### Operating System

Windows 10

### dbt version

1.4.1

### dbt Adapter

Snowflake

### dbt Power User version

0.11.5

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by EFox2413** [comment]
Thanks for posting this! 

I had the same error and it was driving me crazy until I disabled dbt power user and it fixed it.

> PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'dbt_packages\\intercom\\integration_tests'

I'm on dbt Power User v 0.11.6 and dbt version 1.3.1
BigQuery
Windows 11

**Comment by egor-larin** [comment]
Got the same :(

**Comment by Frunkh** [comment]
Yeah we got the same issue here. We have to disable the extension in order to execute dbt deps. We have the following packages:

`packages:
  - package: dbt-labs/codegen
    version: 0.8.1
  - package: dbt-labs/dbt_utils
    version: 1.0.0
  - package: dbt-labs/dbt_project_evaluator
    version: 0.3.0
  - package: calogica/dbt_expectations
    version: 0.8.2
  - package: elementary-data/elementary
    version: 0.7.3`

**Comment by alexnikitchuk** [comment]
the same issue

**Comment by tac14yo** [comment]
Same issue here

**Comment by nguyenann13** [comment]
Same here

**Comment by btello** [comment]
Same.  The only way that I've found to reliably unblock myself is to Ctrl+P and then reload window.  If you run dbt deps *quickly* afterwards it usually works.

**Comment by zachloertscher** [comment]
I'm experiencing the same issue - had to disable dbt power user & reload the window to get `dbt deps` to work

**Comment by anandgupta42** [comment]
@zachloertscher Which version of the extension are using?

**Comment by Frunkh** [comment]
> @zachloertscher Which version of the extension are using?

We're still experiencing the problem with v0.31.3

**Comment by mseveritt-mck** [comment]
This issue still exists on **Windows** machines as also [reported on dbt-core](https://github.com/dbt-labs/dbt-core/issues/4372) in 2021 and is unsolved. The suggested solution on that reported issue is: deactivate vscode-dbt-power-user :-( 

By version **0.52.0** of vscode-dbt-power-user this problem is now **enforced** on each opening of dbt project folders as the extension now executes dbt deps automatically leading to multiple occasions of this error. @mdesmet

Some thoughts how to mitigate the situation:
- Make running dbt deps automatically configurable.
- Run debt deps automatically only when needed and not each time. (if that is possible)

**Comment by eva-aaalife** [comment]
> [@mdesmet](https://github.com/mdesmet) Thanks for the quick fix within version 0.52.1!

Hey @mseveritt-mck  - I'm on version 0.52.3. I saw that it's fixed by @mdesmet but I'm not sure where to find the documentation to config this option?

Would appreciate if you could share how to disable `dbt deps` running automatically at startup after his fix! I tried this page (https://docs.myaltimate.com/setup/optConfig/)  but didn't find it. 

Thanks in advance! 

- Eva

**Comment by mdesmet** [comment]
<img width="735" alt="Image" src="https://github.com/user-attachments/assets/842f0722-5fa4-4c61-accf-5416c2de1f0a" />
